App.js

```diff
@@ -1,26 +1,36 @@
-import React, { Component } from 'react';
-import { createStore, combineReducers, applyMiddleware } from 'redux';
-import { Provider } from 'react-redux';
-import defaultReducer from './Reducers/Reducer';
-import navigation from './Reducers/NavigationReducer';
-import { middleware } from './Navigation/Navigator';
+import React, { Component } from "react";
+import { createStore, combineReducers, applyMiddleware } from "redux";
+import { Provider } from "react-redux";
+import defaultReducer from "./Reducers/Reducer";
+import navigation from "./Reducers/NavigationReducer";
+import settings from "./Reducers/settingsReducer";
+import { middleware } from "./Navigation/Navigator";
 import { translations } from "./Localization/translations";
-import I18n, { i18nState } from "redux-i18n"
-import Navigator from './Navigation/Navigator';
+import I18n, { i18nState } from "redux-i18n";
+import Navigator from "./Navigation/Navigator";
+import thunk from "redux-thunk";
+import { loadSettings } from "./Actions/settingsActions";
+import DeviceInfo from "react-native-device-info";
 
 const appReducer = combineReducers({
   i18nState,
   navigation,
-  defaultReducer,
+  settings,
+  defaultReducer
 });
 
-const store = createStore(
-  appReducer,
-  applyMiddleware(middleware),
-);
+const store = createStore(appReducer, applyMiddleware(middleware, thunk));
 
+let deviceId =
```

Comment:
what do you think about storing device id in store? (pass to reducer on LOAD_SUCCESS)
this will make possible calling storeSettings without second retrieving deviceid on settings screen?